### ISSUE TITLE
Youtube-dl not working properly

### ISSUE THREAD

**Issue by FlipperLP** [rr-/szurubooru#418]
I don't quite know if I am doing something wrong, but it used to work before adding a nginx container to the docker-compose for a [Firefox add-in](https://github.com/neobooru/SzuruChrome): https://github.com/neobooru/SzuruChrome/issues/12

Every time I try to download a YouTube video, I'm presented with the full link from the Google server, it tried to access as the errormessage: 
![image](https://user-images.githubusercontent.com/27019397/124274488-9db25780-db41-11eb-9841-03ecbbf20d61.png)

I am running the newest version of szurubooru on my rpi (self compiled; just updated) and the problem is still there. Can anyone else recreate this problem?

**Comment by neobooru** [comment]
Could you also attach the nginx conf file you are currently using?

**Comment by sgsunder** [comment]
Are you sure that the youtube file you are trying to download does not exceed the `max_dl_filesize` you set in the config?

**Comment by FlipperLP** [comment]
Should I set that in both nginx configs or only mine I added afterwards?

**Comment by sgsunder** [comment]
For youtube-dl you only need to set that in the `config.yaml` file

**Comment by FlipperLP** [comment]
For me it says
`max_dl_filesize: 25.0E+6 # maximum filesize limit in bytes`

**Comment by sgsunder** [comment]
Yeah, that's the default file size (25 MB), which would probably be too small for a typical youtube video. You might want to bump it to 1 GB (`1.0E+9`) or higher if you intend on storing full videos.

**Comment by FlipperLP** [comment]
Ok, that seemed to work, but now I get a `Gateway Time-out`-Error on longer/bigger videos.

**Comment by kgnfth** [comment]
@sgsunder  Why use scientific notation, whats the purpose ?

**Comment by sgsunder** [comment]
@kgnfth The value is stored in bytes. You could type it as `max_dl_filesize: 1000000000` but it's easier to read in scientific notation.  `max_dl_filesize: 1.0E+9` is very clearly 1 GB if you know SI prefixes.

@FlipperLP The file should still upload regardless. The problem is that this application is not really designed for such large files (primarily designed as an imageboard), so it doesn't have any mechanism for upload/download status and it won't respond to the HTTP request until the file is fully downloaded. For large files, this can make it appear that the backend has crashed and the NGINX proxy will respond as such.
